tracing/tracing.go

```diff
@@ -0,0 +1,149 @@
+package tracing
+
+// Copyright 2018 Microsoft Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+import (
+	"context"
+	"net/http"
+	"os"
+
+	"contrib.go.opencensus.io/exporter/ocagent"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/plugin/ochttp/propagation/tracecontext"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+var (
+	// Transport is the default tracing RoundTripper. If tracing is disabled
+	// the default RoundTripper is used which provides no instrumentation.
+	Transport = http.DefaultTransport
+
+	// Enabled is the flag for marking if tracing is enabled.
+	Enabled = false
```

Comment:
Do you expect customers to use this value or is it purely for testing purposes?  If it's the latter then perhaps we can not export it.

Reply:
Or make an `enabled()` func that predicates on `sampler` being nil or not.

Reply:
well, I was thinking they might use it to check if monitoring is enabled on the sdk. if we want we could leave it unexported and if there is a requirement we can export it.

Reply:
I see.  It might be better to make it `Enabled()` then so it's effectively read-only (I could see somebody setting it to false thinking it would turn it off).

Reply:
fair enough